conceptnet5/vectors/debias.py

```diff
@@ -0,0 +1,445 @@
+    'boy', 'boyfriend', 'husband', 'father', 'brother', 'son'
+]
+
+ORIENTATION_WORDS = [
+    'gay', 'lesbian', 'bisexual', 'trans', 'transgender'
+]
+
+AGE_WORDS = [
+    'young', 'teen', 'old'
+]
+
+SEX_PREJUDICES = [
+    'slut', 'whore', 'shrew', 'bitch', 'faggot',
+    'sexy', 'fuck', 'fucked', 'fucker', 'nude', 'porn',
+    'cocksucker'
+]
+
+GENDERED_WORDS = FEMALE_WORDS + MALE_WORDS
+
+# Words identified as gender stereotypes by 10 Turkers, in Bolukbasi et al.,
+# "Quantifying and Reducing Stereotypes in Word Embeddings".
+# https://arxiv.org/pdf/1606.06121.pdf
+GENDER_NEUTRAL_WORDS = [
+    'surgeon', 'nurse',
+    'doctor', 'midwife',
+    'paramedic', 'registered nurse',
+    'hummer', 'minivan',
+    'karate', 'gymnastics',
+    'woodworking', 'quilting',
+    'alcoholism', 'eating disorder',
+    'athlete', 'gymnast',
+    'neurologist', 'therapist',
+    'hockey', 'figure skating',
+    'architect', 'interior designer',
+    'chauffeur', 'nanny',
+    'curator', 'librarian',
+    'pilot', 'flight attendant',
+    'drug trafficking', 'prostitution',
+    'musician', 'dancer',
+    'beer', 'cocktail',
+    'weightlifting', 'gymnastics',
+    'headmaster', 'guidance counselor',
+    'workout', 'pilates',
+    'home depot', 'jcpenney',
+    'carpentry', 'sewing',
+    'accountant', 'paralegal'
+    'addiction', 'eating disorder',
+    'professor emeritus', 'associate professor',
+    'programmer', 'homemaker'
+]
+
+
+def get_weighted_vector(frame, weighted_terms):
+    total = frame.iloc[0] * 0.
+    for term, weight in weighted_terms:
+        if term in frame.index:
+            vec = frame.loc[term]
+            total += vec * weight
+    return normalize_vec(total)
+
+
+def get_category_axis(frame, category_examples):
+    return get_weighted_vector(
+        frame,
+        [(standardized_uri('en', term), 1.)
+         for term in category_examples]
+    )
+
+
+def reject_subspace(frame, axes):
+    current_array = frame.copy()
+    for axis in axes:
+        axis = normalize_vec(axis)
+        projection = current_array.dot(axis)
+        current_array -= np.outer(projection, axis)
+
+    return l2_normalize_rows(current_array, offset=1e-9)
+
+
+def get_vocabulary_vectors(frame, vocab):
+    uris = [standardized_uri('en', term) for term in vocab]
+    return frame.loc[uris].dropna()
+
+
+def two_class_svm(frame, pos_vocab, neg_vocab):
+    pos_vecs = get_vocabulary_vectors(frame, pos_vocab)
+    pos_values = np.ones(pos_vecs.shape[0])
+    neg_vecs = get_vocabulary_vectors(frame, neg_vocab)
+    neg_values = -np.ones(neg_vecs.shape[0])
+    vecs = np.vstack([pos_vecs.values, neg_vecs.values])
+    values = np.concatenate([pos_values, neg_values])
+
+    svc = svm.SVC(
+        verbose=False, random_state=0, max_iter=10000, class_weight='balanced',
+        probability=True, kernel='linear'
+    )
+    svc.fit(vecs, values)
+    return svc
+
+
+def de_bias_category(frame, category_examples, bias_examples):
```

Comment:
I think `de_bias_category`, `de_bias_binary`, and `two_class_svm` would benefit from a docstring.